applications/DEMApplication/python_scripts/DEM_analysis_stage.py

```diff
@@ -4,12 +4,10 @@
 import sys
 from KratosMultiphysics import *
 from KratosMultiphysics.DEMApplication import *
-sys.path.insert(0, '')
-from analysis_stage import AnalysisStage
+from KratosMultiphysics.analysis_stage import AnalysisStage
 from importlib import import_module
 
-# Import MPI modules if needed. This way to do this is only valid when using OpenMPI. For other implementations of MPI it will not work.
-if "OMPI_COMM_WORLD_SIZE" in os.environ or "I_MPI_INFO_NUMA_NODE_NUM" in os.environ:
+if IsDistributedRun():
```

Comment:
@philbucher  Here we had two diferent types of mpi runs:
- distributed domain (the typical mpi run where each node only knows of part of the domain)
- the whole simulation and domain in all nodes (where you run exactly the same case everywhere, but with a different random seed)
I guess IsDistributedRun() is only the first one, right?

Reply:
Hm now I am not sure
do you launch both cases with MPI? Because `IsDistributedRun` is basically telling you if you launched with MPI or not

adding @jcotela to the discussion, since he added this fct

Reply:
`IsDistributedRun()` is ultimately based on mpi detection very similar to the one here, since it is set based on the value of `using_mpi` in the snippet below:
https://github.com/KratosMultiphysics/Kratos/blob/173902d6cd05f97a2200af19dfc75c15ad7d57ff/kratos/python_interface/__init__.py#L31-L44

It is not in principle related to being in one of the two cases of @maceligueta, the only thing it implies is that mpi was initialized at some point.